src/pip/_internal/configuration.py

```diff
@@ -111,7 +111,7 @@ class Configuration(object):
     """
 
     def __init__(self, isolated, load_only=None):
-        # type: (bool, Kind) -> None
+        # type: (bool, Optional[Kind]) -> None
```

Comment:
As described [here](https://stackoverflow.com/a/51084715/1698058), a default of `None` causes mypy to implicitly make the corresponding type `Optional`. Personally I don't like that default and would support enabling stricter options, but just to explain why it didn't alert about this annotation.

Reply:
Thanks, I did not know that it applies to parameter list as well.